homeassistant/helpers/schema_config_entry_flow.py

```diff
@@ -44,7 +44,7 @@ class SchemaFlowFormStep(SchemaFlowStep):
     """
 
     validate_user_input: Callable[
-        [SchemaCommonFlowHandler, dict[str, Any]], dict[str, Any]
+        [SchemaCommonFlowHandler, dict[str, Any]], Coroutine[Any, Any, dict[str, Any]]
```

Comment:
I think it will be easier to manage if we enforce `Awaitable`/`Coroutine`
I don't think there is much benefit in allowing non-async functions.